source/kernel/src/registry/mod.rs

```diff
@@ -62,14 +75,38 @@ pub trait RegisteredDriver {
     /// This is the type of an UNSUCCESSFUL response sent FROM the driver service
     type Error: 'static;
 
+    /// An initial message sent to the service by a client when establishing a
+    /// connection.
+    ///
+    /// This may be used by the service to route connections to specific
+    /// resources owned by that service, or to determine whether or not the
+    /// connection can be established. If the service does not require initial
+    /// data from the client, this type can be set to [`()`].
+    // XXX(eliza): ideally, we could default `Hello` to () and `ConnectError` to
+    // `Infallible`...do we want to do that? it requires a nightly feature.
+    type Hello: 'static;
```

Comment:
Nit, we might want to give this a name like "Introduction" or "ConnectDetails" or something, but we can bikeshed later.

Reply:
i stole the name from TLS handshakes, which start with a  `ClientHello` message followed by a `ServerHello` message. But, if you don't like it, i'm happy to change it later...